pkg/transport/process/processors/example/main.go

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+
+	"github.com/gardener/component-cli/pkg/transport/process"
+	"github.com/gardener/component-cli/pkg/transport/process/extensions"
+)
+
+const processorName = "example-processor"
+
+// a test processor which adds its name to the resource labels and the resource blob.
+// the resource blob is expected to be plain text data.
+func main() {
+	addr := os.Getenv(extensions.ServerAddressEnv)
+
+	if addr == "" {
+		// if addr is not set, use stdin/stdout for communication
+		if err := processorRoutine(os.Stdin, os.Stdout); err != nil {
+			log.Fatal(err)
+		}
+		return
+	}
+
+	h := func(r io.Reader, w io.WriteCloser) {
+		if err := processorRoutine(r, w); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	srv, err := process.NewUDSServer(addr, h)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	srv.Start()
+
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	<-stop
+
+	srv.Stop()
+}
+
+func processorRoutine(inputStream io.Reader, outputStream io.WriteCloser) error {
+	defer outputStream.Close()
+
+	tmpfile, err := ioutil.TempFile("", "")
```

Comment:
is there are reason why we first write to a file instead of in memory/process the stream directly?

Reply:
we cannot directly consume the input stream as a tar archive, because of the tar format. the header at the beginning of the tar archive contains the info about all (!) files in the archive. however these files must first be written. if we directly open the input stream via a tar reader, the reader will be unable to read because the header is incomplete unless all files are written. and once the buffer is full because the reader doesn't read, it will cause the writer to block which will completely block the system.

Reply:
would interest me too